packages/cli/src/commonlib/appStudioLogin.ts

```diff
@@ -73,17 +75,51 @@ export class AppStudioLogin extends login implements AppStudioTokenProvider {
    * Get team access token
    */
   async getAccessToken(showDialog = true): Promise<string | undefined> {
-    if (!AppStudioLogin.codeFlowInstance.account) {
-      const loginToken = await AppStudioLogin.codeFlowInstance.getToken();
-      if (loginToken && AppStudioLogin.statusChange !== undefined) {
-        const tokenJson = await this.getJsonObject();
-        await AppStudioLogin.statusChange("SignedIn", loginToken, tokenJson);
+    let loginToken;
+    try {
+      if (!AppStudioLogin.codeFlowInstance.account) {
+        CliTelemetry.sendTelemetryEvent(TelemetryEvent.AccountLoginStart, {
+          [TelemetryProperty.AccountType]: TelemetryAccountType.M365,
+        });
+        loginToken = await AppStudioLogin.codeFlowInstance.getToken();
+        if (loginToken && AppStudioLogin.statusChange !== undefined) {
+          const tokenJson = await this.getJsonObject();
+          await AppStudioLogin.statusChange("SignedIn", loginToken, tokenJson);
+        }
+        await this.notifyStatus();
+      } else {
+        loginToken = await AppStudioLogin.codeFlowInstance.getToken();
       }
-      await this.notifyStatus();
-      return loginToken;
+
+      if (loginToken) {
+        const array = loginToken.split(".");
+        const buff = Buffer.from(array[1], "base64");
+        const res = JSON.parse(buff.toString("utf-8"));
+        CliTelemetry.sendTelemetryEvent(TelemetryEvent.AccountLogin, {
+          [TelemetryProperty.AccountType]: TelemetryAccountType.M365,
+          [TelemetryProperty.Success]: TelemetrySuccess.Yes,
+          [TelemetryProperty.UserId]: res.oid,
+          [TelemetryProperty.Internal]: res.upn.endsWith("@microsoft.com") ? "true" : "false",
```

Comment:
Is it possible to get machine domain or machine account via node? That may be more accurate to determine an internal user.

Reply:
This logic  is confirmed with PM. For now it's enough for our requirement.